Sources/Transitions/TransitionExtentions.swift

```diff
@@ -0,0 +1,136 @@
+// Copyright © 2019 SkeletonView. All rights reserved.
+
+import UIKit
+
+
+internal extension UIView {
+    @objc func fadeIn(duration:TimeInterval) {
+        fade(duration: duration, fadeIn: true)
+    }
+    
+    @objc func fadeOut(duration:TimeInterval) {
+        fade(duration: duration, fadeIn: false)
+    }
+    
+    fileprivate func fade(duration:TimeInterval, fadeIn: Bool) {
+        guard let sublayers = self.layer.sublayers else {
+            return
+        }
+        
+        for tempLayer in sublayers {
+            handleLayer(layer: tempLayer, fadeIn: fadeIn, duration: duration)
+        }
+        
+        fadeBackgroundColor(duration: duration, fadeIn: fadeIn)
+    }
+    
+    fileprivate func handleLayer(layer:CALayer, fadeIn:Bool, duration:TimeInterval) {
+        if layer.isSkeletonLayer {
+            return
+        }
+        if duration != 0 {
+            fadeLayer(layer: layer, duration: duration, fadeIn: fadeIn)
+        }
+        else {
+            layer.opacity = fadeIn ? 1 : 2
+        }
+    }
+    
+    fileprivate func fadeLayer(layer:CALayer, duration:Double, fadeIn:Bool) {
+        let animation = CABasicAnimation(keyPath: "opacity")
+        animation.fromValue = fadeIn ? 0 : 1
+        animation.toValue = fadeIn ? 1 : 0
+        animation.duration = duration
+        
+        layer.opacity = fadeIn ? 1 : 0
+        layer.add(animation, forKey: nil)
+    }
+    
+    fileprivate func fadeBackgroundColor(duration:Double, fadeIn:Bool) {
+        UIView.transition(with: self, duration: duration, options: .transitionCrossDissolve, animations: {
+            self.backgroundColor = fadeIn ? self.viewState?.backgroundColor : .clear
+        }, completion: nil)
+    }
+}
+
+internal extension UILabel {
+    override func fadeIn(duration:TimeInterval) {
+        if duration == 0 {
+            self.textColor = (self.viewState as? RecoverableTextViewState)?.textColor
```

Comment:
`self.viewState` will never be a `RecoverableTextViewState`. Use `self.labelState` here instead.